vcf.c

```diff
@@ -451,7 +452,8 @@ int bcf_hdr_register_hrec(bcf_hdr_t *hdr, bcf_hrec_t *hrec)
 
     // INFO/FILTER/FORMAT
     char *id = NULL;
-    int type = -1, num = -1, var = -1, idx = -1;
+    uint32_t type = -1, var = -1;
```

Comment:
Yes, in as much as it does what the old code did.  The main change here is that `var` and `type` are masked correctly when packed into `info`.